docs/Contributing.md

```diff
@@ -0,0 +1,84 @@
+# Contributing
+
+This document is designed to act as an outline for the process of contributing to the project. You will find some workflow tips, naming conventions, and other useful information. A standard process will increase productivity.
+
+## Outline
+
+As you begin adding code to the repository you will notice there is a lot of code. And we have a standard naming convention for folders and files that you should follow.
+
+1. Fork the repository, and keep your `master` branch the same as `messari/subgraphs`, before creating a feature branch in your repository **pull** upstream changes.
+2. Create your feature branch `branch-name`.
+3. Once you begin to **push** changes create a Pull Request (PR) in `messari/subgraphs` and set it as a draft until you it is ready for review.
+
+![How to convert to draft](./images/contributing/convert-to-draft.png)
+
+> See "Naming Conventions" for how to name your PR
+
+4. Once your PR is ready for review go ahead and click "Ready for review". As a courtesy it is nice to let the reviewer know that it is ready through a DM.
+
+At this point you will just need to follow the "Reviewing" process until your PR is merged.
+
+## Reviewing
+
+### Reviewee
+
+- Generally you will need to make changes, either respond to a change/comment with a question, rebuttle, or comment. Otherwise, make the change and resolve the issue.
+- When you feel changes are sufficient, let the reviewer know again that the PR is ready for another round of reviews.
+- Do not make a new PR for changes, this will make it harder for the reviewer to track their progress.
+
+### Reviewer
+
+- This part is straight forward, and if you are reviewing you know what to do.
+- It is courtesy to let the developer know that you are done with the review.
+
+> This is an iterative process that takes time, so don't think it will always be easy.
+
+## Rebasing
```

Comment:
It's definitely easier, but I would suggest rebasing when possible.

Merges leave ugly "scar" commits with descriptions like:

`Merge branch 'master' of github.com:messari/subgraphs`

Also, such commits make understanding git history more difficult to follow. Tools like git blame, git log, and git bisect become much more cumbersome, since some changes will appear in the historical tree more than once.

Reply:
Thats a good point. Ive always done it that way but for the sake of keeping things clean ill start rebasing

Reply:
@MichaelC1999 gitLevel++;